site/content/book/v2/helloworldpy.md

```diff
@@ -0,0 +1,170 @@
+                    service_endpoint = s.service_endpoint,
+                    routing_keys = s.routing_keys,
+                    accept = s.accept
+                )
+                for s in did_doc.service
+                if isinstance(s, DIDCommServicePeerDID)
+            ] if did_doc.service else []
+        )
+```
+- **DID Peer creation**
+We add the following function to simplify key generation, key storage, and the creation of a DID Peer:
+```
+async def create_simple_peer_did() -> str:
+    agreem_keys = generate_x25519_keys_as_jwk_dict()
+    auth_keys = generate_ed25519_keys_as_jwk_dict()
+    did = peer_did.create_peer_did_numalgo_2(
+        [VerificationMaterialAgreement(
+                type = VerificationMethodTypeAgreement.JSON_WEB_KEY_2020,
+                format = VerificationMaterialFormatPeerDID.JWK,
+                value = agreem_keys[1])
+        ],
+        [VerificationMaterialAuthentication(
+                type = VerificationMethodTypeAuthentication.JSON_WEB_KEY_2020,
+                format = VerificationMaterialFormatPeerDID.JWK,
+                value = auth_keys[1])
+        ],
+        None
+    )
+    did_doc = DIDDocPeerDID.from_json(peer_did.resolve_peer_did(did))
+    pk = auth_keys[0]
+    pk["kid"] = did_doc.auth_kids[0]
+    await secrets_resolver.add_key(jwk_to_secret(pk))
+    private_key = agreem_keys[0]
+    private_key["kid"] = did_doc.agreement_kids[0]
+    await secrets_resolver.add_key(jwk_to_secret(private_key))
+
+    return did
+```
+This function creates a basic DID Peer with only one *Agreement* key, one *Authentication* key, and no *Service* part. You can find more options in the library documentation.
+
+### Step 4: Create DIDs
+Using our `create_simple_peer_did` helper function, Alice and Bob can create their DID Peer that they will share and use when communicating privately between each other:
+```
+alice_did = await create_simple_peer_did()
+print("Alice's DID:", alice_did)
+bob_did = await create_simple_peer_did()
+print("Bob's DID:", bob_did)
+```
+You should get something similar to:
+```
+Alice's DID: did:peer:2.Ez6LSt4Jscr227NFyuzKHT85haVE4AFVXm1tDwYeZ5xenxMmW.Vz6MkfvwnoNS6Cto38MEMbqdnypVDN7gS4oAMaHFkjAUse5JE
+Bob's DID: did:peer:2.Ez6LSetXDUvD8rBSei5TU5ew7VRdWyNBr5mAsxr6EoHFTxt9f.Vz6Mkty3Nu98rnrHfk1GBCurF7EFKY5Vb34FAJNJCMhwzduk3
+```
+Remember that while creating these simple DIDs, our helper function also stores the private keys in the `secrets_resolver`. In a real implementation, Alice will have her own secure store in her own wallet, and Bob will have a separated secure store in his own wallet.
+Also, those Peer DIDs can be resolved into DID documents that contain the *Authentication* and *Agreement* public keys.
+
+### Step 5: Encrypt and pack the message
+Alice can create a simple "Hello World" message with:
+```
+message = Message(
+    body = {"msg": "Hello World"},
+    id = "unique-id-24160d23ed1d",
+    type = "my-protocol/1.0",
+    frm = alice_did,
```

Comment:
`from` is defined in the spec however `frm` is imposed by the library, probably due to some internal conflict. I'll make a comment on the description to clarify that difference from the spec.

Reply:
Yes, "from" is a reserved word in python, so we can't use it as a variable or parameter name. Anywhere that the writeup is using `frm` as a python identifier, `frm` is correct. But anywhere that it is using `frm` as the name of a header (maybe nowhere?) it would be wrong. I think adding a note about this (just a sentence) would be good.

Reply:
new commit pushed with clarifications